lib/galaxy/tools/parameters/output_collect.py

```diff
@@ -23,9 +24,99 @@
 log = logging.getLogger( __name__ )
 
 
+class NullToolProvidedMetadata(object):
+
+    def get_new_datasets(self, output_name):
+        return []
+
+    def get_new_dataset_meta_by_basename(self, output_name, basename):
+        return {}
+
+
+class LegacyToolProvidedMetadata(object):
+
+    def __init__(self, job_wrapper, meta_file):
+        self.job_wrapper = job_wrapper
```

Comment:
There is already unit test coverage of this in test_collect_primary_datasets.py and there are several tool tests covering both tool provided metadata classes. Hopefully that is enough?